### ISSUE TITLE
`variable-for-property` not working as expected with linear-gradient

### ISSUE THREAD

**Issue by GabrielCousin** [sasstools/sass-lint#1096]
Hello 👋 
I don't think I have done something wrong but I am currently facing this issue.

**What version of Sass Lint are you using?**
1.10.2

**Please include any relevant parts of your configuration**
```
  variable-for-property:
    - 2
    - properties:
        - color
        - background
        - background-color
```
**What did you do? Please include the actual source code causing the issue.**
```
.foo {
    background: linear-gradient(to right, $gray-lightest 0%, $gray-lightest 35%, $concrete 35%, $concrete 100%);
}
```

**What did you expect to happen?**
I think this particular line is respecting the lint rules

**What actually happened? Please include any error messages given to you by Sass Lint.**
```
13:3  error    Values for properties of type 'background' may only be variables  variable-for-property
```
**If you're using a IDE plugin have you tried the CLI too?**
Both CLI and Atom plugin

**Comment by DanPurdy** [comment]
This is actually working as expected and specified in the [docs](https://github.com/sasstools/sass-lint/blob/develop/docs/rules/variable-for-property.md) Basically you're not using a variable as a value for the background property here, you're just passing variables to the linear-gradient function.

By the letter of the rule to make this lint correctly you would do something like 

```scss
$my-gradient: linear-gradient(to right, $gray-lightest 0%, $gray-lightest 35%, $concrete 35%, $concrete 100%);

.foo {
    background: $my-gradient;
}
```

I'm going to close the issue as it's not a bug, but if you have a suggestion on how to change this rule or an idea about it then please feel free to open a new issue with your proposal.
Thanks! 👍
